app/src/main/java/ch/epfl/sdp/Map/MapFragment.java

```diff
@@ -67,10 +67,16 @@
 
     @VisibleForTesting
     public MapboxMap getMap() {
+        while (map == null) {
+            getMap();
+        }
         return map;
     }
 
     public PathsHandler getPathsHandler() {
+        while (pathsHandler == null) {
```

Comment:
What is this ? :)
Can't you add a timer to check if you have not been waiting for too long ?

Reply:
Oh I forgot to remove this, thanks!

Reply:
xD